provisioning/miniprov/hare_mp/cdf.py

```diff
@@ -64,22 +100,22 @@ def _create_pool_descriptions(self) -> List[PoolDesc]:
                 data_devices_count += len(
                     conf.get(
                         f'server_node>{node}>storage>cvg[0]>data_devices'))
+            pool_type = self._get_pool_type(cluster_id, x)
```

Comment:
@knekrasov, How about instead of extracting a pool_type from conf-store, we iterate over supported pool types and fetch the information for the ones available in the conf-store?
SNS pool is a must have, an exception must be raised is not present. 
DIX is also required as per the latest discussions but for now let's not have it as a mandatory pool type. If none of the pool types are present, an exception is raised.
What do you think?

Reply:
@mssawant If you look into `_get_pool_type()`, its algorithm is quite the same to what you described.

Reply:
Not really, because there can be multiple pool types in a storage set, e.g. SNS, DIX, _get_pool_type() returns just 1 pool type per storage_set. We need to iterate over the pool_types and get the information for all the available pool_types in a  given storage set.

Reply:
According to @ujjwalpl storage set = pool. One pool can have only one type.

Even if we imagine that a storage set that has types, say sns and dix, then we possibly can understand it as "I need to create two pools". But how will we map which drive will belong to which pool?

Reply:
So, there are data devices and metadata devices. Data devices belong to data pool, SNS. Metadata devices belong to metadata pool, in this case dix. SNS layout parameters, data, parity and spare apply only to the SNS pool. Important property of the layout parameters is that the total number of corresponding devices in the pool need to be >= data + parity + spare.
Thus SNS layout parameters cannot apply to DIX pool (comprising of metadata devices) in a storage set. Thus we have different layout parameters defined for DIX pool type corresponding to the metadata devices in the pool.
Presently the default layout parameters for DIX pool are 1 + 0 + 0 in Motr.
Thus, storage set (pool) comprises of 2 pools, data and metadata. Both will have different layout parameters.

Reply:
Thanks @mssawant for explanation, now the picture looks pretty different to me.

Reply:
@mssawant Your comment is addressed, please see.

@SwapnilGaonkar7 In the last two commits I refactored your code a bit, so it would be nice if you can review.

Reply:
@knekrasov , tried your branch, added following dix pool layout params,
```
          "durability": {
            "sns": {
              "data": "4",
              "parity": "2",
              "spare": "2"
            },
            "dix": {
              "data": "1",
              "parity": "1",
              "spare": "1"
            }
          },
```
getting error as,
```
[root@ssc-vm-1623:root] /opt/seagate/cortx/hare/bin/hare_setup config --config 'json:///opt/seagate/cortx/hare/conf/hare.config.conf.tmpl.3-node.sample' --file /root/config_cluster.cdf
Error performing configuration (Command ['cfgen', '-o', '/var/lib/hare', '/root/config_cluster.cdf'] exited with error code 1. Command output: Traceback (most recent call last):
  File "/opt/seagate/cortx/hare/bin/cfgen", line 1775, in <module>
    main()
  File "/opt/seagate/cortx/hare/bin/cfgen", line 133, in main
    validate_cluster_desc(cluster_desc)
  File "/opt/seagate/cortx/hare/bin/cfgen", line 245, in validate_cluster_desc
    validate_pools_desc(desc['pools'])
  File "/opt/seagate/cortx/hare/bin/cfgen", line 349, in validate_pools_desc
    f"{name}: disk_refs are only supported for pools of 'sns' type"
AssertionError: storage1__dix: disk_refs are only supported for pools of 'sns' type
)
```
with just sns things are working fine.

Reply:
Hm, cfgen code has interesting `XXX` comment:
```python
        if 'disk_refs' not in pool:
            continue

        # XXX Do we want to support disk_refs for DIX and MD pools as well?
        assert t is PoolT.sns, \
            f"{name}: disk_refs are only supported for pools of 'sns' type"
        disk_refs = pool['disk_refs']
        # Allowing diskrefs to be empty until the CDF generation is fixed.
        if disk_refs:
            assert disk_refs, \
                f'Pool {name!r}: disk_refs must not be empty' \
                ' (it can be omitted though)'
            assert all_unique(repr(x) for x in disk_refs), \
                f'Pool {name!r}: disk_refs must be unique'
```

@mssawant Are you sure that DIX pool can (and should) have disk refs?
cc @s-arya, @suvratjoshi 

In the meantime I'll add one more commit to unblock this behaviour in cfgen...

Reply:
Commit that unblocks DIX in cfgen has been added. I tested the change locally at a single node, bootstrap with dix and sns pool in the profile was successful. @mssawant please review.

Reply:
>@mssawant Are you sure that DIX pool can (and should) have disk refs?

I don't see a reason why should we restrict disk_refs for DIX pools as we have metadata devices.

Reply:
Yep, works, verified.